app/src/androidTest/java/ch/epfl/sweng/erpa/activities/UserProfileActivityTest.java

```diff
@@ -0,0 +1,131 @@
+package ch.epfl.sweng.erpa.activities;
+
+import android.content.Intent;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.junit.runner.RunWith;
+
+import ch.epfl.sweng.erpa.ErpaApplication;
+import ch.epfl.sweng.erpa.R;
+import ch.epfl.sweng.erpa.model.UserProfile;
+import ch.epfl.sweng.erpa.operations.RemoteServicesProviderCoordinator;
+import ch.epfl.sweng.erpa.services.UserProfileService;
+import ch.epfl.sweng.erpa.services.dummy.DummyRemoteServicesProvider;
+import ch.epfl.sweng.erpa.services.dummy.database.DummyUserService;
+import toothpick.Scope;
+import toothpick.Toothpick;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static ch.epfl.sweng.erpa.util.TestUtils.getUserProfile;
+import static junit.framework.TestCase.assertTrue;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.core.AllOf.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class UserProfileActivityTest {
+
+    private ActivityTestRule<UserProfileActivity> activityTestRule = new ActivityTestRule<>(UserProfileActivity.class, false, false);
+    private ErpaApplication app;
+    private Scope scope;
+    private UserProfile userProfile = getUserProfile("hey");
+    private UserProfileService ups;
+    @Rule
+    public final TestRule chain = RuleChain.outerRule(
+            new ExternalResource() {
+                @Override
+                protected void before() throws Throwable {
+                    app = (ErpaApplication) InstrumentationRegistry.getTargetContext().getApplicationContext();
+                    scope = Toothpick.openScope(app);
+                    RemoteServicesProviderCoordinator rsp = scope.getInstance(RemoteServicesProviderCoordinator.class);
+                    rsp.rspClassFromFullyQualifiedName(DummyRemoteServicesProvider.class.getName()).ifPresentOrElse(
+                            rsp::bindRemoteServicesProvider, () -> {
+                                throw new RuntimeException("Couldn't find RSP");
+                            });
+                    ups = new DummyUserService(InstrumentationRegistry.getTargetContext().getApplicationContext());
+                }
+
+                @Override
+                protected void after() {
+                    Toothpick.reset(scope);
+                    app.installModules(scope);
+                }
+            }
+
+    ).around(activityTestRule);
+    @Before
+    public void createIntent()
```

Comment:
Please move the opening bracket right of the closing parenthesis on all the methods of this class